src/widget/wcoverartmenu.cpp

```diff
@@ -82,6 +83,64 @@ void WCoverArtMenu::slotChange() {
     coverInfo.source = CoverInfo::USER_SELECTED;
     coverInfo.coverLocation = selectedCoverPath;
     coverInfo.setImage(image);
+
+    if (m_pConfig->getValue<bool>(mixxx::library::prefs::kCreateCopyOfTheCoverArtConfigKey)) {
+        QString savedCoverArtLocation = fileInfo.absoluteDir().path() + "/" +
+                fileInfo.completeBaseName() + ".jpg";
+        if (QFile::exists(savedCoverArtLocation)) {
+            QMessageBox saveBox(
+                    QMessageBox::Question,
+                    tr("Overwrite Existing Cover Art?"),
+                    tr("The track already has a related Cover Art in same the path\n"
+                       "Do you want to overwrite it?\n"
+                       "Cover art located at: %1")
+                            .arg(fileInfo.absoluteDir().path()));
+
+            saveBox.setDefaultButton(QMessageBox::Yes);
+            saveBox.addButton(tr("&Overwrite"), QMessageBox::AcceptRole);
+            saveBox.addButton(tr("&Update without Overwrite"), QMessageBox::ApplyRole);
+            saveBox.addButton(tr("&Discard"), QMessageBox::DestructiveRole);
+
+            switch (saveBox.exec()) {
+            case QMessageBox::DestructiveRole:
+                qDebug() << "Cover art is not updated";
+                return;
+
+            case QMessageBox::ApplyRole:
+                qDebug() << "Cover art is updated but didn't overwrite the existing cover art";
+                qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
+                emit coverInfoSelected(coverInfo);
+                return;
+
+            case QMessageBox::AcceptRole:
+                QFile::remove(savedCoverArtLocation);
+                qDebug() << "Updated and overwritten";
+                break;
+            default:
+                break;
+            }
+        }
+
+        else {
+            if (m_pConfig->getValue<bool>(mixxx::library::prefs::
+                                kInformCoverArtLocationConfigKey)) {
+                QMessageBox::information(nullptr,
+                        tr("Cover Art Location"),
+                        tr("Cover Art saved at \n%1").arg(fileInfo.absoluteDir().path()));
+            }
+        }
+
+        QByteArray coverArtByteArray;
```

Comment:
All file operations should be done in a separate thread, to not stall the waveforms or such during disk access.

Reply:
You ma take a look at TrackExportWorker how we can do it.